### PR TITLE
Flatpak fixes

### PR DESCRIPTION
**Pull request by hadess**
A couple of fixes I used to create a Flatpak for testing your latest fixes.

### CODE REVIEW

File: flatpak/org.gnome.FeedReader.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
         "--share=ipc",
         /* Access to Wayland */
         "--socket=wayland",
+        /* Access to OpenGL */
+        "--device=dri",
```
Comment (jscurtu) [inline flatpak/org.gnome.FeedReader.json]: Why?

**Comment by jscurtu** [comment]
Why allow access to DRI, can you explain this?
A script already exist, its the make file ..

**Comment by hadess** [comment]
> Why allow access to DRI, can you explain this?

It's in the commit message.

> A script already exist, its the make file ..

There's no "Makefile" in the repository, there's a CMakeLists.txt file, which needs to be processed, to create the Makefile, to then allow creating the Flatpak.

Would be nice not to close the PR before I have a chance to answer your questions.

**Comment by jscurtu** [comment]
The make file is here:
https://github.com/jangernert/FeedReader/tree/master/flatpak

It does not need to be processed after Cmake.

**Comment by hadess** [comment]
The other changes still apply, and would stop this error:
libEGL warning: wayland-egl: could not open /dev/dri/card0 (No such file or directory)
when running under Wayland, and allow WebKit to use GL for compositing.

**Comment by hadess** [comment]
Actually, it is WebKit creating a Wayland compositor that throws this error:
```
libEGL warning: wayland-egl: could not open /dev/dri/card0 (No such file or directory)
WaylandCompositor requires eglCreateImage and eglDestroyImage.
Nested Wayland compositor could not initialize EGL
```

**Comment by hadess** [comment]
Also, there are bugs or misunderstandings in the Makefile.
```
BRANCH    = master
VERSION   = $(BRANCH)-` date +"%Y%m%d" `

dist:
       flatpak build-bundle repo/ $(BUNDLE).flatpak org.gnome.$(PACKAGE) $(BRANCH)
```

I doubt that's the intended behaviour. The BRANCH above I guess is supposed to be the git branch. But that's not used anywhere, and the json file will always point to the git master branch. The BRANCH below, in the `flatpak build-bundle` command is an ostree branch. It's always master if you don't use branches in ostree.

**Comment by hadess** [comment]
I've updated the branch to take the location of the Makefile into account. Do I need to open a new PR, or could this one be reopened?

**Comment by jscurtu** [comment]
you have to create a new pull request as the branch was repushed or recreated.

**Comment by jscurtu** [comment]
I am not completely satisfied with the Makefile though.. But the base is ok.. 
The way I see it, if is not x86_64 will still create the filename with "x86_64" even if 32bit detected. 
Also, even though there is a $branch you use static master on creating the bundle.

**Comment by jscurtu** [comment]
There are more things that need to be discussed, why was the branch removed from the builder json file? I think it should stay there.. 
If you cant properly sed it and replace it in the Makefile then it needs to be rethink'ed.

Or is there a different reason?

**Comment by jscurtu** [comment]
after looking what other webkit apps like GNOME Web do, access to dri seems legit and accepted.

**Comment by jscurtu** [comment]
Please create a new PR for: 

flatpak: Allow access to OpenGL 
flatpak: Add comments for Wayland and network permissions

and I will accept those...

The others need a little more work, which is why it was so basic in the first place:
- rework the logic of the Makefile 

The following is not very well placed and formatted it also needs to be redone. 
- README: Add instructions on how to build Flatpak bundle

**Comment by hadess** [comment]
> you have to create a new pull request as the branch was repushed or recreated

If this PR was still opened, it would automatically be updated with my updated branch.

**Comment by hadess** [comment]
> The way I see it, if is not x86_64 will still create the filename with "x86_64" even if 32bit detected.

I don't see how it would do that.

Anyway, I'll wait until you reopen this to comment further.

**Comment by bil-elmoussaoui** [comment]
@hadess We canno't reopen the PR, github shows an insensitive button with "the flatpak-fixes-1 branch was forced-pushed or recreated"

**Pull request closed without merging**